ada_feeding_perception/ada_feeding_perception/food_on_fork_logistic_reg_training.py

```diff
@@ -0,0 +1,50 @@
+import pandas as pd
+import numpy as np
+from sklearn.linear_model import LogisticRegression
+import joblib
+
+
+def main():
+    csv_to_read_from = "../datasets/Ross_7-11-23.csv"
+
+    # get the data frame
+    df = pd.read_csv(csv_to_read_from)
+
+    # shuffle all the rows
+    df = df.sample(frac=1)
+
+    # split the dataset (60% train/20% validation/20% test)
+    # Note: the first split occurs at 60%; the second split occurs at 80% and they are all stored in the variables
+    train, validation, test = np.split(df, [int(0.6 * len(df)), int(0.8 * len(df))])
```

Comment:
You never actually use the validation or test sets, right? It seems like you are unnecessarily reducing the size of the training set. 

You should do the following:

1. Run cross-validation, so we actually have a number of how it performs on unseen data, and print that number. See here for an example of [running cross validation](https://scikit-learn.org/stable/modules/cross_validation.html#computing-cross-validated-metrics).
2. Then, after running cross-validation, use the entire set as the train set to re-train the model, and save that. This is better because: (a) it gives you more train data; and (b) it makes the trained model deterministic -- `logistic_reg_model.pkl` will be the exact same every time you run this file.